osutil/mountentry_linux.go

```diff
@@ -26,6 +26,46 @@ import (
 	"syscall"
 )
 
+// This is needed to properly split options that could contain an escaped
```

Comment:
So the original's `SplitMountOptions` intention was solely to split `systemd-mount` options arguments in a way that it accounts for commas as the current logic won't work for systemd-mount options for overlayfs mounts as it simply splits on comma: 
https://github.com/canonical/snapd/blob/6d1aacd0f9b16e2dd3054df21c7bd74185575989/cmd/snap-bootstrap/initramfs_systemd_mount_test.go#L281 

`systemd-mount` 's way of doing this can be found in https://github.com/systemd/systemd/blob/4ec630bfba02543af204f13ad9f5ce4a2329a166/src/basic/extract-word.c#L20

After I moved it under `osutil/mountentry_linux.go`, I also used it to split mount options in `ParseMountEntry()` but indeed that should be done on a per-fs basis properly. So to clarify things a bit:
1. I kept `splitMountOptions()` as it was originally introduced; a helper function that accounts for escaped commas when splitting an options string. 
2. Added 2 new public functions with names to indicate their intended usage. They are both simply invoking the helper but it should make things a bit clearer in potential future refactors. `SplitSystemdMountOptions()` with a link to the `systemd-mount` parsing logic which aims to mimic. `SplitOverlayfsMountOptions()` which I will be using in `ParseMountEntry()` only for when an "overlay" type is detected. In any other case the previous logic will be followed. I added a TODO about this for a follow-up PR.

Reply:
Also looking at Valentin's comment in https://github.com/canonical/snapd/pull/14179#discussion_r1711294497 I think my current code doesn't accurately follows the kernel parsing logic so I am inclined to revert to the old behavior for `ParseMountEntry` and remove `SplitOverlayMountOptions()` completely and address this in a separate PR. As far as I can tell `ParseMountEntry` is not used in the context of this PR.